image-loader/app/controllers/ImageLoaderController.scala

```diff
@@ -65,8 +96,122 @@ class ImageLoaderController(auth: Authentication,
 
   def index: Action[AnyContent] = AuthenticatedAndAuthorised { indexResponse }
 
-  def quarantineOrStoreImage(uploadRequest: UploadRequest)(implicit logMarker: LogMarker) = {
-    quarantineUploader.map(_.quarantineFile(uploadRequest)).getOrElse(uploader.storeFile(uploadRequest))
+  private def quarantineOrStoreImage(uploadRequest: UploadRequest)(implicit logMarker: LogMarker) = {
+    quarantineUploader.map(_.quarantineFile(uploadRequest)).getOrElse(for { uploadStatusUri <- uploader.storeFile(uploadRequest)} yield{uploadStatusUri.toJsObject})
+  }
+
+  private def handleMessageFromIngestBucket(sqsMessage:SQSMessage)(implicit logMarker: LogMarker): Future[Unit] = {
+
+    logger.info(logMarker,sqsMessage.toString)
+
+    parseS3DataFromMessage(sqsMessage) match {
+      case Left(failString) =>
+        logger.warn(logMarker, s"Failed to parse s3 data from SQS message : $failString")
+        Future.unit
+      case Right(s3data) =>
+        val s3IngestObject = S3IngestObject(s3data, store)
+        val approximateReceiveCount = getApproximateReceiveCount(sqsMessage)
+        logMarker ++ Map(
+          "uploadedBy" -> s3IngestObject.uploadedBy,
+          "uploadedTime" -> s3IngestObject.uploadTime,
+          "contentLength" -> s3IngestObject.contentLength,
+          "filename" -> s3IngestObject.filename,
+          "isUiUpload" -> s3IngestObject.maybeMediaIdFromUiUpload.isDefined,
+        )
+
+        if (approximateReceiveCount > 2) {
+          val errorMessage = s"File processing has been attempted $approximateReceiveCount times. Moving to fail bucket."
+          logger.warn(logMarker, errorMessage)
+          store.moveObjectToFailedBucket(s3IngestObject.key)
+
+          s3IngestObject.maybeMediaIdFromUiUpload match {
+            case Some(imageId) => uploadStatusTable.updateStatus( //FIXME use set status to avoid potential ConditionNotMet (when status table rows have expired/TTL)
+              imageId, UploadStatus(StatusType.Failed, Some(errorMessage))
+            )
+            case None => Future.unit
+          }
+        }
+
+        attemptToProcessIngestedFile(s3IngestObject).transformWith {
```

Comment:
attemptToProcessIngestedFile is another function which can throw up to the stream and presumably crash it

Reply:
stream crashing is handled with a recover (which now catches Throwable rather than Exception) and worst case if the future/source does ever exit then healthcheck will fail and box will cycle

Reply:
> I think this wants to be in an else block? otherwise cases where `if (approximateReceiveCount > 2) {` will continue and run this step as well, which will presumably always fail

good point about this though - will address that now